bitshuffle/library.py

```diff
@@ -0,0 +1,231 @@
+'''Usage-independent functions'''
+
+from __future__ import division, generators, print_function, absolute_import
+
+import re
+import bz2
+import gzip
+import string
+import base64
+from hashlib import sha256
+from sys import stderr, stdin, exit as exit_with_code
+from .errors import LEVELS, ERRORS
+
+VERSION = '0.0.1-git'
+DEBUG = False
+COMPATLEVEL = "1"
+DEFAULT_MSG = "This is encoded with BitShuffle, which you can download" + \
+    " from https://github.com/charlesdaniels/bitshuffle "
+
+try:  # does nothing if python3
+    assert gzip.compress
+    assert gzip.decompress
+except AttributeError:  # python2
+    import io
+    # taken straight from gzip.py
+    # pylint: disable=invalid-name
+
+    def gzip_compress(data, compresslevel=5):
+        # pylint: disable=missing-docstring
+        buf = io.BytesIO()
+        c = compresslevel
+        with gzip.GzipFile(fileobj=buf, mode='wb', compresslevel=c) as f:
+            f.write(data)
+        return buf.getvalue()
+
+    def gzip_decompress(data):
+        # pylint: disable=missing-docstring
+        with gzip.GzipFile(fileobj=io.BytesIO(data)) as f:
+            return f.read()
+
+    gzip.compress = gzip_compress
+    gzip.decompress = gzip_decompress
+    del gzip_compress, gzip_decompress
+
+
+def warn(integer, *argv, **kwargs):
```

Comment:
This does not need to be handled for #60, but in the long run, we should probably use python's `logging` library. When we started with Bitshuffle, it was sufficiently small that logging would be overkill, but at this point, we want to make sure we aren't polluting other developer's output streams.

Reply:
added https://github.com/charlesdaniels/bitshuffle/issues/62